Clarify the default behavior of value tag (#49)
pelotom/unionize: README.md

## Example

Provide `unionize` with a mapping of tags to value types:

```ts
  import { unionize, ofType } from 'unionize';

  const Actions = unionize({
    ADD_TODO: ofType<{ id: string; text: string }>(),
    CLEAR_TODOS: {}, // For "empty" types, just use {}

>   });

Reviewer: Unindent this block?

```

Extract the inferred tagged union:

```ts
import { UnionOf } from 'unionize'

type Action = UnionOf<typeof Actions>;
```

The inferred type is:

```ts
type Action = ({
    tag: "ADD_TODO";
} & {
    id: string;
    text: string;
}) | {
    tag: "CLEAR_TODOS";
}
```

> We can also optionally override the tag and/or value property names: 

Reviewer: Since before this point there was no notion of a "value" property, it doesn't really make sense to talk about overriding it. Maybe just

```suggestion
We can also optionally provide tag and/or value property names:
```

>   | { type: CLEAR_TODOS; payload: {} };

Reviewer: This is a preexisting condition, but I just realized the literals `ADD_TODO`, etc. aren't quoted here and they should be.

```

which is now [FSA compliant](https://github.com/redux-utilities/flux-standard-action).

> Note that when the tag property is omitted, `"tag"` will be used as the tag property. But if the value tag name is omitted, the generated sum type does not use a default `"value"` property -- Instead, it intersects the value type with `{[tagType]: tag }`.

Reviewer: ```suggestion
Note that when the tag property is omitted, `"tag"` will be used as the tag property. But if the value tag name is omitted, the generated sum type does not use a default `"value"` property -- Instead, it intersects the value type with `{ [tagProp]: tag }`.
```

> While this is convenient while dealing with value types which are objects (No need to access the value through a `.value` accessor), it should be obvious that the value type can only be an object type if value tag is omitted. 

Reviewer: ```suggestion
While this is convenient when dealing with value types which are objects (No need to access the value through a `.value` accessor), it should be obvious that the value type can only be an object type if value tag is omitted. 
```

> So something like `ofType<'SHOW_ALL' | 'SHOW_ACTIVE' | 'SHOW_COMPLETED'>` is not legal as a value type.

Reviewer: ```suggestion
So something like `ofType<'SHOW_ALL' | 'SHOW_ACTIVE' | 'SHOW_COMPLETED'>` is only legal as a value type when a `value` property name is provided:
```

```ts
const Actions = unionize({
  ADD_TODO: ofType<{ id: string; text: string }>(),
  SET_VISIBILITY_FILTER: ofType<                           
    "SHOW_ALL" | "SHOW_ACTIVE" | "SHOW_COMPLETED"
  >()
})
```

> The above fails with the error: 

Reviewer: ```suggestion
The above fails with the error
```
